<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/ionicons-2.0.1/css/ionicons.min.css" rel="stylesheet" />
<link href="https://fonts.googleapis.com/css?family=Nunito+Sans|Roboto|Lobster" rel="stylesheet">

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 64px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h2 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h3 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h4 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h5 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h6 {
  padding-top: 69px;
  margin-top: -69px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">



<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">
      	<img alt="Nemabiome Logo" src="images/logo/nemabiome_compact_64.png" style="width:35px;height:35px">
      </a>
    </div>
    	
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
				
				<li>
					<a href="overview.html">
						<span class="ion ion-map"></span>
    				Overview
					</a>
				</li>
	
			<!-- Parasite -->
				<li>
  				<a href="parasite.html">
    				<span class="ion ion-bug"></span>
    				Parasite Prep
     			</a>
				</li>
				
			<!-- Sequencing -->	
				<li>
  				<a href="sequencing.html">
    				<span class="ion ion-erlenmeyer-flask"></span>
    				Sequencing
     			</a>
				</li>
				
			<!-- Analysis -->
				<li>
  				<a href="analysis.html">
    				<span class="ion ion-monitor"></span>
    				Analysis
     			</a>
				</li>
			
			<!-- Interpretation -->
				<li>
  				<a href="interpretation.html">
    				<span class="ion ion-stats-bars"></span>
    				Interpretation
     			</a>
				</li>

			</ul>
			
			<!-- Citation and about -->
	    <ul class="nav navbar-nav navbar-right">

				<li>
  				<a href="citation.html">
    				<span class="ion ion-information-circled"></span>
    		    Citation
  				</a>
				</li>
				
			
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div id="analysis-protocols" class="section level1 tabset tabset-pills tabset-fade">
<h1>Analysis Protocols</h1>
<p>Once you have generated the data from the MiSeq you must complete the following steps in order to analyze the data.</p>
<div id="software-requirements" class="section level2">
<h2>Software Requirements</h2>
<p>The primary software used for analysis is Mothur. This program is command line based, and will require you to type simple commands into either Terminal (Mac) or Command Prompt (Windows).</p>
<p>To download Mothur go to the Mothur github <a href="https://github.com/mothur/mothur/releases/latest">downloads</a> page and download the latest version for your operating system. Unzip then move this folder to your home directory (your C: Drive on a PC, or your ‘User’ folder on a Mac).</p>
<div class="panel panel-info">
<div class="panel-heading">
Tip
</div>
<div class="panel-body">
<p>To check that you have placed the folder in the correct location, open up either Terminal (Mac) or Command Prompt (PC). Type <code>~/mothur/mothur</code> (Mac) or <code>C:\mothur\mothur.exe</code> (PC). If your screen looks similar to this:</p>
<div class="figure">
<img src="images/mothur_screenshot.png" />

</div>
<p>You have done it correctly.</p>
</div>
</div>
</div>
<div id="preparing-files-for-analysis" class="section level2">
<h2>Preparing files for analysis</h2>
<div id="downloading-sequencing-data-files-from-basespace" class="section level3">
<h3>Downloading sequencing data files from BaseSpace</h3>
<p>To download the sequencing fields from BaseSpace. Go to <a href="https://basespace.illumina.com/home/index" class="uri">https://basespace.illumina.com/home/index</a>, log-in, and navigate to your sequencing run. Click “Download” and select “All FASTQ files for this run”. Then click “Download your files”. BaseSpace uses the “BaseSpace Downloader” to download your files. If you have not previously downloaded this program, you must download it at this time. Select the location to download your files.</p>
<p>Once your files have downloaded, navigate to the folder where the files were downloaded. There will be one folder that corresponds to each sample that was sequenced. In each of these folders there will be two files, one that corresponds to the forward reads, and one that corresponds to the reverse reads. The forward and reverse read files for all samples will need to be moved to a single folder for analysis to conduct the analysis.</p>
<div class="panel panel-success">
<div class="panel-heading">
Recommendation
</div>
<div class="panel-body">
<p>Create a folder on your home directory with the name of your sequencing run (for the rest of this example it will be called “MiSeq_Run”). Move all of your files into this folder.</p>
</div>
</div>
<p>Once the files have been moved, the files must be unzipped/uncompressed. This can be done with the Archive Utility on a Mac and any file compression software, such as WinZip, on a PC. Make sure you unzip your files into the folder that you have created (i.e. “Miseq_Run”)</p>
</div>
<div id="create-sample-list" class="section level3">
<h3>Create sample list</h3>
<p>You will need to create a file that contains the name of your sample, and specifies which files correspond to this sample. This will tell the program which files to analyze for each particular sample. You can download the file “nemabiome.files” in the <a href="interpretation.html">Examples</a> section for an example of what this file must look like. Each line in this file represents a single sample that must be analyzed. The first column is the sample name, the second column is the file name for the forward read file, and the third column is the name of the reverse read file. Each column is separated by a single tab. An example is also listed below.</p>
<p>Example:</p>
<pre><code>sample1 sample1_S1_L001_R1_001.fastq     sample1_S1_L001_R2_001.fastq
sample2 sample2_S2_L001_R1_001.fastq     sample2_S2_L001_R2_001.fastq
sample3 sample3_S3_L001_R1_001.fastq     sample3_S3_L001_R2_001.fastq
...</code></pre>
<p>Sample names may vary between different Illumina platforms, so please check the names of your downloaded files, to ensure the file names are correct. Please name this file as “stability.files”, with the “.files” file extension. Place this file in your folder that you created for analysis. This is the only file that you will need to create from scratch for this analysis, if you use the provided files.</p>
</div>
<div id="download-analysis-files" class="section level3">
<h3>Download analysis files</h3>
<p>Download the analysis files <a href="downloads/nematode_ITS2_database_version1_3.fasta" download>Nematode ITS2 database version 1.3</a>, <a href="downloads/nematode_taxonomy_1_3.tax" download>Nematode taxonomy file (version 1.3)</a> and <a href="downloads/batchfile.txt" download>Mothur batch file</a>. Place these files in the folder that contains your sequencing data (i.e. ‘MiSeq_Run’)</p>
<p>For a standard application do not make any changes to these files. If you are aware of the number of processors that your computer has, open up the <code>batchfile.txt</code> file, go to the first line and change the parameter “processors” from “processors=2” to the number of processors in your computer.</p>
</div>
</div>
<div id="running-analysis" class="section level2">
<h2>Running analysis</h2>
<p>Once all files have been downloaded, open up Terminal or Command Prompt. You will then need to navigate to the folder you created, which can be done using the “cd” command. If you created your analysis folder in your home directory type:</p>
<pre><code>cd MiSeq_Run</code></pre>
<p>This will change to this folder.</p>
<p>Then type:</p>
<pre><code>~/mothur/mothur batchfile.txt</code></pre>
<p>(On a Mac)</p>
<p>OR</p>
<pre><code>C:\mothur\mothur.exe C:\MiSeq_Run\batchfile.txt</code></pre>
<p>(On a PC)</p>
<p>This will launch Mothur, and begin to complete the commands that are contained within the batchfile. You will need to leave your computer open and running while the analysis completes. The amount of time that it will take to run will vary with your computer, and the number of samples/reads that you will be analyzing. General run times will probably be around ~4 hours, on a standard computer, 96 samples with an average read depth of ~50,000 reads.</p>
<p>Once the analysis has completed, you will have a file called <code>nemabiome_results.summary</code>. You can open this file up and copy the contents into an Excel workbook to properly view the data.</p>
</div>
<div id="dada2-example-workflow" class="section level2">
<h2>DADA2 Example Workflow</h2>
<p>The following is a sample analysis using the DADA2 pipeline (<a href="https://benjjneb.github.io/dada2/tutorial.html" class="uri">https://benjjneb.github.io/dada2/tutorial.html</a>). This is meant to be an example and not a definitive workflow. First we load packages and set a seed for reproducibility.</p>
<pre class="r"><code>library(DECIPHER)</code></pre>
<pre><code>## Loading required package: Biostrings</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:parallel&#39;:
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter,
##     Find, get, grep, grepl, intersect, is.unsorted, lapply, Map,
##     mapply, match, mget, order, paste, pmax, pmax.int, pmin,
##     pmin.int, Position, rank, rbind, Reduce, rownames, sapply,
##     setdiff, sort, table, tapply, union, unique, unsplit, which,
##     which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## 
## Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     strsplit</code></pre>
<pre><code>## Loading required package: RSQLite</code></pre>
<pre class="r"><code>packageVersion(&quot;DECIPHER&quot;)

library(dada2)</code></pre>
<pre><code>## Loading required package: Rcpp</code></pre>
<pre><code>## Registered S3 methods overwritten by &#39;ggplot2&#39;:
##   method         from 
##   [.quosures     rlang
##   c.quosures     rlang
##   print.quosures rlang</code></pre>
<pre class="r"><code>packageVersion(&quot;dada2&quot;) 

library(ShortRead)</code></pre>
<pre><code>## Loading required package: BiocParallel</code></pre>
<pre><code>## Loading required package: Rsamtools</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: GenomicAlignments</code></pre>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## Loading required package: DelayedArray</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:Biobase&#39;:
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## 
## Attaching package: &#39;DelayedArray&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>## The following object is masked from &#39;package:Biostrings&#39;:
## 
##     type</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     aperm, apply, rowsum</code></pre>
<pre class="r"><code>packageVersion(&quot;ShortRead&quot;) 

library(Biostrings)
packageVersion(&quot;Biostrings&quot;)

library(ggplot2)
packageVersion(&quot;ggplot2&quot;)

set.seed(106)</code></pre>
<p>To use the DADA2 pipeline, primers must be removed from amplicons. This can be done in R using cutadapt. Here we specify the path of sequences to be processed (using “/Miseq_Run” from previous examples) and the path of Cutadapt on your machine (we used “/home/gilleardlab/anaconda3/bin/cutadapt”). You will also need to specify the primers you used as the “FWD” and “REV” variables. We start off by creating all possible orientations of the forward and reverse sequences.</p>
<pre><code>##  [1] &quot;cutadapt&quot;                     &quot;filtN&quot;                       
##  [3] &quot;G1_S1_L001_R1_001.fastq.gz&quot;   &quot;G1_S1_L001_R2_001.fastq.gz&quot;  
##  [5] &quot;G10_S10_L001_R1_001.fastq.gz&quot; &quot;G10_S10_L001_R2_001.fastq.gz&quot;
##  [7] &quot;G11_S11_L001_R1_001.fastq.gz&quot; &quot;G11_S11_L001_R2_001.fastq.gz&quot;
##  [9] &quot;G12_S12_L001_R1_001.fastq.gz&quot; &quot;G12_S12_L001_R2_001.fastq.gz&quot;
## [11] &quot;G13_S13_L001_R1_001.fastq.gz&quot; &quot;G13_S13_L001_R2_001.fastq.gz&quot;
## [13] &quot;G14_S14_L001_R1_001.fastq.gz&quot; &quot;G14_S14_L001_R2_001.fastq.gz&quot;
## [15] &quot;G15_S15_L001_R1_001.fastq.gz&quot; &quot;G15_S15_L001_R2_001.fastq.gz&quot;
## [17] &quot;G16_S16_L001_R1_001.fastq.gz&quot; &quot;G16_S16_L001_R2_001.fastq.gz&quot;
## [19] &quot;G17_S17_L001_R1_001.fastq.gz&quot; &quot;G17_S17_L001_R2_001.fastq.gz&quot;
## [21] &quot;G18_S18_L001_R1_001.fastq.gz&quot; &quot;G18_S18_L001_R2_001.fastq.gz&quot;
## [23] &quot;G19_S19_L001_R1_001.fastq.gz&quot; &quot;G19_S19_L001_R2_001.fastq.gz&quot;
## [25] &quot;G2_S2_L001_R1_001.fastq.gz&quot;   &quot;G2_S2_L001_R2_001.fastq.gz&quot;  
## [27] &quot;G20_S20_L001_R1_001.fastq.gz&quot; &quot;G20_S20_L001_R2_001.fastq.gz&quot;
## [29] &quot;G21_S21_L001_R1_001.fastq.gz&quot; &quot;G21_S21_L001_R2_001.fastq.gz&quot;
## [31] &quot;G22_S22_L001_R1_001.fastq.gz&quot; &quot;G22_S22_L001_R2_001.fastq.gz&quot;
## [33] &quot;G23_S23_L001_R1_001.fastq.gz&quot; &quot;G23_S23_L001_R2_001.fastq.gz&quot;
## [35] &quot;G24_S24_L001_R1_001.fastq.gz&quot; &quot;G24_S24_L001_R2_001.fastq.gz&quot;
## [37] &quot;G25_S25_L001_R1_001.fastq.gz&quot; &quot;G25_S25_L001_R2_001.fastq.gz&quot;
## [39] &quot;G26_S26_L001_R1_001.fastq.gz&quot; &quot;G26_S26_L001_R2_001.fastq.gz&quot;
## [41] &quot;G3_S3_L001_R1_001.fastq.gz&quot;   &quot;G3_S3_L001_R2_001.fastq.gz&quot;  
## [43] &quot;G4_S4_L001_R1_001.fastq.gz&quot;   &quot;G4_S4_L001_R2_001.fastq.gz&quot;  
## [45] &quot;G5_S5_L001_R1_001.fastq.gz&quot;   &quot;G5_S5_L001_R2_001.fastq.gz&quot;  
## [47] &quot;G6_S6_L001_R1_001.fastq.gz&quot;   &quot;G6_S6_L001_R2_001.fastq.gz&quot;  
## [49] &quot;G7_S7_L001_R1_001.fastq.gz&quot;   &quot;G7_S7_L001_R2_001.fastq.gz&quot;  
## [51] &quot;G8_S8_L001_R1_001.fastq.gz&quot;   &quot;G8_S8_L001_R2_001.fastq.gz&quot;  
## [53] &quot;G9_S9_L001_R1_001.fastq.gz&quot;   &quot;G9_S9_L001_R2_001.fastq.gz&quot;</code></pre>
<pre><code>##                Forward             Complement                Reverse 
## &quot;ACGTCTGGTTCAGGGTTGTT&quot; &quot;TGCAGACCAAGTCCCAACAA&quot; &quot;TTGTTGGGACTTGGTCTGCA&quot; 
##                RevComp 
## &quot;AACAACCCTGAACCAGACGT&quot;</code></pre>
<p>Next we will remove all sequences containing ambiguous bases, and count the number of times our primers are observed in the remaining sequences.</p>
<pre><code>##                  Forward Complement Reverse RevComp
## FWD.ForwardReads   25344          0       0       0
## FWD.ReverseReads       0          0       0       0
## REV.ForwardReads       0          0       0       0
## REV.ReverseReads   21503          0       0       0</code></pre>
<p>Now we can use cutadapt to remove primers, and check that they have been successfully removed.</p>
<pre class="r"><code>cutadapt &lt;- &quot;/home/stefan/.local/bin/cutadapt&quot; # CHANGE ME to the cutadapt path on your machine
system2(cutadapt, args = &quot;--version&quot;) # Run shell commands from R

#now creating output filenames for cutadapted files, and define parameters to give cutadapt command
#critical parameters are the primers and they need to be in the right orientation

path.cut &lt;- file.path(path, &quot;cutadapt&quot;)
if(!dir.exists(path.cut)) dir.create(path.cut)
fnFs.cut &lt;- file.path(path.cut, basename(fnFs))
fnRs.cut &lt;- file.path(path.cut, basename(fnRs))  #cutadapt not working -&gt; due to files not gz?

FWD.RC &lt;- dada2:::rc(FWD)
REV.RC &lt;- dada2:::rc(REV)
# Trim FWD and the reverse-complement of REV off of R1 (forward reads)
R1.flags &lt;- paste(&quot;-g&quot;, FWD, &quot;-a&quot;, REV.RC) 
# Trim REV and the reverse-complement of FWD off of R2 (reverse reads)
R2.flags &lt;- paste(&quot;-G&quot;, REV, &quot;-A&quot;, FWD.RC) 
# Run Cutadapt
for(i in seq_along(fnFs)) {
  system2(cutadapt, args = c(R1.flags, R2.flags, &quot;-n&quot;, 2, # -n 2 required to remove FWD and REV from reads
                             &quot;-o&quot;, fnFs.cut[i], &quot;-p&quot;, fnRs.cut[i], # output files
                             fnFs.filtN[i], fnRs.filtN[i])) # input files
}

# remember to check things - can check presence of primers in the first cutadapted sample 
rbind(FWD.ForwardReads = sapply(FWD.orients, primerHits, fn = fnFs.cut[[1]]), 
      FWD.ReverseReads = sapply(FWD.orients, primerHits, fn = fnRs.cut[[1]]), 
      REV.ForwardReads = sapply(REV.orients, primerHits, fn = fnFs.cut[[1]]), 
      REV.ReverseReads = sapply(REV.orients, primerHits, fn = fnRs.cut[[1]]))</code></pre>
<p>Now that primers are removed, we can begin the DADA2 pipeline. We start with plotting quality profiles of our sequences to determine suitable quality filtering parameters.</p>
<pre><code>##  [1] &quot;G1_S1_L001&quot;   &quot;G10_S10_L001&quot; &quot;G11_S11_L001&quot; &quot;G12_S12_L001&quot;
##  [5] &quot;G13_S13_L001&quot; &quot;G14_S14_L001&quot; &quot;G15_S15_L001&quot; &quot;G16_S16_L001&quot;
##  [9] &quot;G17_S17_L001&quot; &quot;G18_S18_L001&quot; &quot;G19_S19_L001&quot; &quot;G2_S2_L001&quot;  
## [13] &quot;G20_S20_L001&quot; &quot;G21_S21_L001&quot; &quot;G22_S22_L001&quot; &quot;G23_S23_L001&quot;
## [17] &quot;G24_S24_L001&quot; &quot;G25_S25_L001&quot; &quot;G26_S26_L001&quot; &quot;G3_S3_L001&quot;  
## [21] &quot;G4_S4_L001&quot;   &quot;G5_S5_L001&quot;   &quot;G6_S6_L001&quot;   &quot;G7_S7_L001&quot;  
## [25] &quot;G8_S8_L001&quot;   &quot;G9_S9_L001&quot;</code></pre>
<p>For the purposes of our analysis, we chose to conduct quality filtering with maximum expected errors of 2 for the forward sequence, 5 for the reverse, to truncate after a quality score of 2 or lower, and to filter out amplicons with lengths shorter than 50 base pairs. Again, filters for your own pipeline may differ. After quality filtering, we can run our sequences through learnErrors to create an error model, and then dereplicate our sequences.</p>
<pre><code>##                              reads.in reads.out
## G1_S1_L001_R1_001.fastq.gz      28069     25002
## G10_S10_L001_R1_001.fastq.gz    30391     27055
## G11_S11_L001_R1_001.fastq.gz    26771     24258
## G12_S12_L001_R1_001.fastq.gz    28793     25794
## G13_S13_L001_R1_001.fastq.gz    28812     26158
## G14_S14_L001_R1_001.fastq.gz    33590     31499</code></pre>
<pre><code>##                              reads.in reads.out
## G1_S1_L001_R1_001.fastq.gz      28069     25002
## G10_S10_L001_R1_001.fastq.gz    30391     27055
## G11_S11_L001_R1_001.fastq.gz    26771     24258
## G12_S12_L001_R1_001.fastq.gz    28793     25794
## G13_S13_L001_R1_001.fastq.gz    28812     26158
## G14_S14_L001_R1_001.fastq.gz    33590     31499
## G15_S15_L001_R1_001.fastq.gz    26882     25290
## G16_S16_L001_R1_001.fastq.gz    28561     26653
## G17_S17_L001_R1_001.fastq.gz    26693     24744
## G18_S18_L001_R1_001.fastq.gz    27489     25525
## G19_S19_L001_R1_001.fastq.gz    28710     25830
## G2_S2_L001_R1_001.fastq.gz      28680     25407
## G20_S20_L001_R1_001.fastq.gz    26774     24102
## G21_S21_L001_R1_001.fastq.gz    28060     25687
## G22_S22_L001_R1_001.fastq.gz    30512     28070
## G23_S23_L001_R1_001.fastq.gz    26251     23650
## G24_S24_L001_R1_001.fastq.gz    24075     22083
## G25_S25_L001_R1_001.fastq.gz    30804     27836
## G26_S26_L001_R1_001.fastq.gz    27777     21986
## G3_S3_L001_R1_001.fastq.gz      30407     27917
## G4_S4_L001_R1_001.fastq.gz      27779     25207
## G5_S5_L001_R1_001.fastq.gz      35661     32700
## G6_S6_L001_R1_001.fastq.gz      32204     29821
## G7_S7_L001_R1_001.fastq.gz      30276     28046
## G8_S8_L001_R1_001.fastq.gz      26885     24639
## G9_S9_L001_R1_001.fastq.gz      22455     20541</code></pre>
<pre><code>## 100272261 total bases in 436807 reads from 17 samples will be used for learning the error rates.</code></pre>
<pre><code>## 100154673 total bases in 436807 reads from 17 samples will be used for learning the error rates.</code></pre>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis

## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<p><img src="analysis_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G1_S1_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5232 unique sequences from 25002 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G10_S10_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 6326 unique sequences from 27055 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G11_S11_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 6039 unique sequences from 24258 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G12_S12_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 4602 unique sequences from 25794 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G13_S13_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5201 unique sequences from 26158 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G14_S14_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5727 unique sequences from 31499 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G15_S15_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5875 unique sequences from 25290 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G16_S16_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 7082 unique sequences from 26653 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G17_S17_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 6133 unique sequences from 24744 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G18_S18_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 6249 unique sequences from 25525 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G19_S19_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5496 unique sequences from 25830 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G2_S2_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5681 unique sequences from 25407 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G20_S20_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 6041 unique sequences from 24102 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G21_S21_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 6168 unique sequences from 25687 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G22_S22_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 4610 unique sequences from 28070 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G23_S23_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 6640 unique sequences from 23650 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G24_S24_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5175 unique sequences from 22083 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G25_S25_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 6443 unique sequences from 27836 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G26_S26_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5719 unique sequences from 21986 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G3_S3_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5780 unique sequences from 27917 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G4_S4_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5610 unique sequences from 25207 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G5_S5_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5650 unique sequences from 32700 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G6_S6_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 7026 unique sequences from 29821 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G7_S7_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5377 unique sequences from 28046 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G8_S8_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5675 unique sequences from 24639 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G9_S9_L001_R1_001.fastq.gz</code></pre>
<pre><code>## Encountered 5228 unique sequences from 20541 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G1_S1_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 16218 unique sequences from 25002 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G10_S10_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 19693 unique sequences from 27055 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G11_S11_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 17647 unique sequences from 24258 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G12_S12_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 17386 unique sequences from 25794 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G13_S13_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 17229 unique sequences from 26158 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G14_S14_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 20973 unique sequences from 31499 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G15_S15_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 17507 unique sequences from 25290 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G16_S16_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 18676 unique sequences from 26653 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G17_S17_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 17774 unique sequences from 24744 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G18_S18_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 18617 unique sequences from 25525 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G19_S19_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 18784 unique sequences from 25830 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G2_S2_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 16294 unique sequences from 25407 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G20_S20_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 19700 unique sequences from 24102 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G21_S21_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 18500 unique sequences from 25687 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G22_S22_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 19470 unique sequences from 28070 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G23_S23_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 18173 unique sequences from 23650 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G24_S24_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 16476 unique sequences from 22083 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G25_S25_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 19593 unique sequences from 27836 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G26_S26_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 15743 unique sequences from 21986 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G3_S3_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 19543 unique sequences from 27917 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G4_S4_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 17698 unique sequences from 25207 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G5_S5_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 20321 unique sequences from 32700 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G6_S6_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 19925 unique sequences from 29821 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G7_S7_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 18122 unique sequences from 28046 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G8_S8_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 16517 unique sequences from 24639 total sequences read.</code></pre>
<pre><code>## Dereplicating sequence entries in Fastq file: ~/MiSeq_Run//cutadapt/filtered/G9_S9_L001_R2_001.fastq.gz</code></pre>
<pre><code>## Encountered 14416 unique sequences from 20541 total sequences read.</code></pre>
<p>After dereplication, we can denoise our sequences using the DADA algorithm. We will also merge our paired-end sequences, allowing for up to 1 mismatch in the overlap region and returning all sequences whether they merged or not.</p>
<pre class="r"><code>dadaFs&lt;-dada(derepFs, err=errF, multithread = TRUE)
dadaRs&lt;-dada(derepRs, err=errR, multithread=TRUE)

mergers&lt;-mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE, maxMismatch = 1, returnRejects = TRUE)</code></pre>
<pre><code>## 21476 paired-reads (in 39 unique pairings) successfully merged out of 21586 (in 57 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 21945 paired-reads (in 31 unique pairings) successfully merged out of 22414 (in 74 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 19465 paired-reads (in 41 unique pairings) successfully merged out of 20198 (in 122 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 21511 paired-reads (in 21 unique pairings) successfully merged out of 21700 (in 41 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 21967 paired-reads (in 42 unique pairings) successfully merged out of 22204 (in 69 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 26347 paired-reads (in 41 unique pairings) successfully merged out of 26549 (in 64 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 20624 paired-reads (in 34 unique pairings) successfully merged out of 21157 (in 95 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 21829 paired-reads (in 49 unique pairings) successfully merged out of 22523 (in 134 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 19821 paired-reads (in 33 unique pairings) successfully merged out of 20156 (in 71 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 20346 paired-reads (in 31 unique pairings) successfully merged out of 20826 (in 125 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 20683 paired-reads (in 61 unique pairings) successfully merged out of 21071 (in 96 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 21762 paired-reads (in 43 unique pairings) successfully merged out of 22044 (in 72 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 17938 paired-reads (in 28 unique pairings) successfully merged out of 18422 (in 54 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 20305 paired-reads (in 56 unique pairings) successfully merged out of 21119 (in 147 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 22812 paired-reads (in 17 unique pairings) successfully merged out of 22891 (in 31 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 18534 paired-reads (in 39 unique pairings) successfully merged out of 19183 (in 88 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 17335 paired-reads (in 37 unique pairings) successfully merged out of 17965 (in 70 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 22173 paired-reads (in 41 unique pairings) successfully merged out of 22490 (in 76 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 18065 paired-reads (in 39 unique pairings) successfully merged out of 18185 (in 59 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 22088 paired-reads (in 42 unique pairings) successfully merged out of 23486 (in 167 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 20751 paired-reads (in 40 unique pairings) successfully merged out of 21169 (in 85 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 27881 paired-reads (in 33 unique pairings) successfully merged out of 28071 (in 58 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 24592 paired-reads (in 68 unique pairings) successfully merged out of 25461 (in 208 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 23582 paired-reads (in 36 unique pairings) successfully merged out of 23839 (in 62 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 20804 paired-reads (in 39 unique pairings) successfully merged out of 21123 (in 85 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<pre><code>## 16965 paired-reads (in 34 unique pairings) successfully merged out of 17326 (in 74 pairings) input.</code></pre>
<pre><code>## Duplicate sequences in merged output.</code></pre>
<p>Before we can assign taxonomy, we can construct a sequence table, remove chimeras using the removeBimeraDenovo command and then create a table of how many sequences were retained after each processing step.</p>
<pre><code>## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.</code></pre>
<pre><code>## [1]  26 208</code></pre>
<pre><code>## Identified 144 bimeras out of 208 input sequences.</code></pre>
<pre><code>## 
## 273 274 281 283 287 288 289 290 291 292 319 421 
##   2   2  12   3   4   4   8   2  18   7   1   1</code></pre>
<pre><code>## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.
## Duplicate sequences detected and merged.</code></pre>
<pre><code>##              input filtered denoisedF denoisedR merged nonchim
## G1_S1_L001   28069    25002     24953     21629  21586   20483
## G10_S10_L001 30391    27055     26921     22528  22414   20281
## G11_S11_L001 26771    24258     24182     20259  20198   18804
## G12_S12_L001 28793    25794     25738     21744  21700   21003
## G13_S13_L001 28812    26158     26083     22264  22204   21187
## G14_S14_L001 33590    31499     31439     26586  26549   24953</code></pre>
</div>
<div id="downloads" class="section level2">
<h2>Download list</h2>
<ul>
<li><a href="downloads/batchfile.txt" download="downloads/batchfile.txt">Mothur batch file</a></li>
<li><a href="downloads/nematode_taxonomy_1_3.tax" download>Nematode taxonomy file (version 1.3)</a></li>
<li><a href="downloads/nematode_ITS2_database_version1_3.fasta" download>Nematode ITS2 database version 1.3</a></li>
</ul>
</div>
</div>

   </div> <!-- articleBandContent -->
</div> <!-- pageContent -->
 

<footer class="footer">
	<nav class="navbar navbar-default navbar-fixed-bottom">
	  <div class="container">
	    <div class="col-sm-12 text-center navbar-text">
	        <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">
    				<img alt="Creative Commons Licence" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" />
    		</a>
    		This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
	    </div>
	  </div>
	</nav>
</footer>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
