<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/ionicons-2.0.1/css/ionicons.min.css" rel="stylesheet" />
<link href="https://fonts.googleapis.com/css?family=Nunito+Sans|Roboto|Lobster" rel="stylesheet">

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 64px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 69px;
  margin-top: -69px;
}

.section h2 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h3 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h4 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h5 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h6 {
  padding-top: 69px;
  margin-top: -69px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->





<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">
      	<img alt="Nemabiome Logo" src="images/logo/nemabiome_compact_64.png" style="width:35px;height:35px">
      </a>
    </div>
    	
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
				
				<li>
					<a href="overview.html">
						<span class="ion ion-map"></span>
    				Overview
					</a>
				</li>
	
			<!-- Parasite -->
				<li>
  				<a href="parasite.html">
    				<span class="ion ion-bug"></span>
    				Parasite Prep
     			</a>
				</li>
				
			<!-- Sequencing -->	
				<li>
  				<a href="sequencing.html">
    				<span class="ion ion-erlenmeyer-flask"></span>
    				Sequencing
     			</a>
				</li>
				
			<!-- Analysis -->
				<li>
  				<a href="analysis.html">
    				<span class="ion ion-monitor"></span>
    				Analysis
     			</a>
				</li>
			
			<!-- Interpretation -->
				<li>
  				<a href="interpretation.html">
    				<span class="ion ion-stats-bars"></span>
    				Interpretation
     			</a>
				</li>

			</ul>
			
			<!-- Citation and about -->
	    <ul class="nav navbar-nav navbar-right">

				<li>
  				<a href="citation.html">
    				<span class="ion ion-information-circled"></span>
    		    Citation
  				</a>
				</li>
				
			
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div id="example-and-interpretation" class="section level1 tabset tabset-pills tabset-fade">
<h1>Example and Interpretation</h1>
<div id="example-dataset" class="section level2">
<h2>Example dataset</h2>
<p>We have provided a worked example that will hopefully help beginners get started with this assay.</p>
<p>This dataset contains 26 pooled strongyle worm samples collected from Brazilian cattle farms. Each sample contains 1000 L3 larvae cultured from fecal samples. The Nemabiome assay was used to investigate species proportions within these pooled samples. The raw results from the assay are usually downloaded directly from the Illumina website (Basespace). We have provided the raw results from these samples below. Begin by downloading and unzipping all three files.</p>
<div id="example-miseq-data-files" class="section level3">
<h3>Example Miseq data files</h3>
<ul>
<li><a href="downloads/Fastq_files_set1.zip" download>Set 1</a></li>
<li><a href="downloads/Fastq_files_set2.zip" download>Set 2</a></li>
<li><a href="downloads/Fastq_files_set3.zip" download>Set 3</a></li>
</ul>
<p>Also download the example file <a href = "downloads/nemabiome.files" download>nemabiome.files</a>. This file contains a list of the samples that we are analyzing, and specifies which files belong to each sample. From this you can generate your stability.files that should look like this.</p>
<p>Use these files to work through the steps set-out under the Analysis tab.</p>
</div>
<div id="example-output" class="section level3">
<h3>Example output</h3>
<p>You can download the file <a href = "downloads/example_nemabiome_results.txt" download>example_nemabiome_results.txt</a>. This is what your output file should look like if you have completed the analysis correctly.</p>
</div>
</div>
<div id="initial-interpretation-of-results" class="section level2">
<h2>Initial interpretation of results</h2>
<p>A youtube video outlines the main steps involved with the initial manipulation of raw read count data and basic visualisation of the example dataset.</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/OkTVcnbskTw" frameborder="0" allowfullscreen>
</iframe>
<p><br></p>
<div id="visualize-raw-read-counts-in-excel-spreadsheet" class="section level3">
<h3>1. Visualize raw read counts in excel spreadsheet</h3>
<p>The first step is to copy and paste the raw read counts into an excel spreadsheet. The fisrt three columns (rankID, taxon, daughterlevel) indicate the level of taxonomic classification. The fourth column (total) is the total number of reads assigned to that taxonomic level. The individual read counts for each sample begins in the 5th column and the sample name is given in the top row. To desplay reads soley assigned at the species level, delete all rows other than “taxlevel 12”.</p>
</div>
<div id="filter-raw-read-count-data" class="section level3">
<h3>2. Filter raw read count data</h3>
<p>Delete rows that have very few total reads. These are very likely the result of background contamination. It is good standard practice to run blank sequencing controls alongside your experimental samples that can be used to assess the level of backgound contamination. In addition, rows that are “unclassied” at the species level also need to looked at critically. If the total read count for these unclassified species are high levels they may represent a species that is not present in the database. if the total read count is lower then the reads are probably the result of PCR/sequencing error and therefore needs removing from the dataset.</p>
</div>
<div id="correction-for-pcr-bias" class="section level3">
<h3>3. Correction for PCR bias</h3>
<p>Not all species of worms amplify at the rate even though they may be present in the same proportions (see Avramenko et al., 2015 for a more comprehensive discussion of the issues involved). Thus correction factors have been estimated that can be used to correct for this amplification bias. Simply multiply the raw read count by the correction factor for each individual species.</p>
<div id="correction-factors" class="section level4">
<h4>Correction factors</h4>
<div id="cattle" class="section level5">
<h5><strong>Cattle</strong></h5>
<table>
<thead>
<tr class="header">
<th align="left">Parsite species</th>
<th align="left">Correction factor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><em>Cooperia punctata</em></td>
<td align="left">2.7087</td>
</tr>
<tr class="even">
<td align="left"><em>Cooperia oncophora</em></td>
<td align="left">1.1545</td>
</tr>
<tr class="odd">
<td align="left"><em>Haemonchus placei</em></td>
<td align="left">0.9339</td>
</tr>
<tr class="even">
<td align="left"><em>Nematodirus helvetianus</em></td>
<td align="left">0.7127</td>
</tr>
<tr class="odd">
<td align="left"><em>Ostertagia ostertagi</em></td>
<td align="left">1.2592</td>
</tr>
</tbody>
</table>
</div>
<div id="sheep" class="section level5">
<h5><strong>Sheep</strong></h5>
<table>
<thead>
<tr class="header">
<th align="left">Parsite species</th>
<th align="left">Correction factor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><em>Cooperia curticei</em></td>
<td align="left">1.6107</td>
</tr>
<tr class="even">
<td align="left"><em>Teladorsagia circumcincta</em></td>
<td align="left">1.1885</td>
</tr>
<tr class="odd">
<td align="left"><em>Trichostrongylus vitrinus</em></td>
<td align="left">1.0239</td>
</tr>
<tr class="even">
<td align="left"><em>Trichostrongylus axei</em></td>
<td align="left">0.9647</td>
</tr>
<tr class="odd">
<td align="left"><em>Trichostrongylus colubriformis</em></td>
<td align="left">1.0239</td>
</tr>
<tr class="even">
<td align="left"><em>Haemonchus contortus</em></td>
<td align="left">0.6970</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="calculate-relative-abundance-of-each-species" class="section level3">
<h3>4. Calculate relative abundance of each species</h3>
<p>To estimate the relative abundance of each species in a sample simply divide the corrected read count by the total corrected read count for each individual sample and multiply by 100.</p>
</div>
<div id="graphical-visualisation-of-species-proportions" class="section level3">
<h3>5. Graphical visualisation of species proportions</h3>
<p>The resulting percentages can then be turned into a stacked bar chart as the the first step at visualising your results.</p>
</div>
</div>
</div>

   </div> <!-- articleBandContent -->
</div> <!-- pageContent -->
 

<footer class="footer">
	<nav class="navbar navbar-default navbar-fixed-bottom">
	  <div class="container">
	    <div class="col-sm-12 text-center navbar-text">
	        <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">
    				<img alt="Creative Commons Licence" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" />
    		</a>
    		This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
	    </div>
	  </div>
	</nav>
</footer>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
